extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -91,17 +91,42 @@ export class AmpIframe extends AMP.BaseElement {
   }
 
   /**
+   * Transforms the src attribute. When possible, it adds `#amp=1` fragment
+   * to indicate that the iframe is running in AMP environment.
+   * @param {?string} src
+   * @return {string|undefined}
+   * @private
+   */
+  transformSrc_(src) {
+    if (!src) {
+      return;
+    }
+    const url = parseUrl(src);
+    // data-URLs are not modified.
+    if (url.protocol == 'data:') {
+      return src;
+    }
+    // If fragment already exists, it's not modified.
+    if (url.hash && url.hash != '#') {
+      return src;
+    }
+    // Add `#amp=1` fragment.
+    return removeFragment(src) + '#amp=1';
```

Comment:
It could be empty, as in `https://example.com/#` - I have that case in the tests.